Mathlib/Data/Set/Image.lean

```diff
@@ -290,6 +290,12 @@ theorem image_congr' {f g : α → β} {s : Set α} (h : ∀ x : α, f x = g x)
   image_congr fun x _ => h x
 #align set.image_congr' Set.image_congr'
 
+/-- Variant of `image_congr`, for one function being the identity. -/
+theorem image_congr'' {α β : Type*} {f : α → β} {g : β → α} {s : Set α}
+    (h : ∀ x : α, x ∈ s → (g ∘ f) x = x) : g ∘ f '' s = s := by
+  rw [image_congr h]
```

Comment:
You're stating this lemma for a composition `g ∘ f` through an intermediate type beta, but wouldn't it be true for any function from alpha to itself? It should probably be written with `EqOn f id s`.

Reply:
Oh, indeed. You're right; I've generalised this and moved it to Data/Set/Function.lean.